Ginger/GingerCoreNETUnitTest/LinuxTransformationTests/ActExcelTests.cs

```diff
@@ -0,0 +1,200 @@
+            actExcel.ExcelActionType = ActExcel.eExcelActionType.ReadData;
+            actExcel.AddNewReturnParams = true;
+            actExcel.SelectAllRows = true;
+
+            //Act
+            actExcel.Execute();
+
+            //Assert
+            Assert.AreEqual(actExcel.ActReturnValues.Count, 4);
+        }
+        [TestMethod]
+        public void ReadCellDataExcelOneCellTest()
+        {
+            //Arrange            
+            ActExcel actExcel = new ActExcel();
+            actExcel.AddOrUpdateInputParamValueAndCalculatedValue("ExcelFileName",
+                TestResources.GetTestResourcesFile(@"Excel" + Path.DirectorySeparatorChar + "Names.xlsx"));
+            actExcel.AddOrUpdateInputParamValueAndCalculatedValue("SheetName", "Sheet1");
+            actExcel.AddOrUpdateInputParamValueAndCalculatedValue("SelectRowsWhere", "A2");
+            actExcel.ExcelActionType = ActExcel.eExcelActionType.ReadCellData;
+            actExcel.AddNewReturnParams = true;
+
+            //Act
+            actExcel.Execute();
+
+            //Assert
+            Assert.AreEqual(actExcel.ActReturnValues.Count, 1);
+        }
+        [TestMethod]
+        public void ReadCellDataExcelAllCellTest()
+        {
+            //Arrange            
+            ActExcel actExcel = new ActExcel();
+            actExcel.AddOrUpdateInputParamValueAndCalculatedValue("ExcelFileName",
+                TestResources.GetTestResourcesFile(@"Excel" + Path.DirectorySeparatorChar + "Names.xlsx"));
+            actExcel.AddOrUpdateInputParamValueAndCalculatedValue("SheetName", "Sheet1");
+            actExcel.ExcelActionType = ActExcel.eExcelActionType.ReadCellData;
+            actExcel.AddNewReturnParams = true;
+            actExcel.SelectAllRows = true;
+
+            //Act
+            actExcel.Execute();
+
+            //Assert
+            Assert.AreEqual(actExcel.ActReturnValues.Count, 12);
+        }
+        [TestMethod]
+        public void WriteExcelOneRowWithPKTest()
+        {
+            //Arrange            
+            ActExcel actExcel = new ActExcel();
+            actExcel.RunOnBusinessFlow = new GingerCore.BusinessFlow();
+            actExcel.AddOrUpdateInputParamValueAndCalculatedValue("ExcelFileName",
+                TestResources.GetTestResourcesFile(@"Excel" + Path.DirectorySeparatorChar + "Names.xlsx"));
+            actExcel.AddOrUpdateInputParamValueAndCalculatedValue("SheetName", "Sheet1");
+            actExcel.AddOrUpdateInputParamValueAndCalculatedValue("SelectRowsWhere", "Last='Cohen'");
+            actExcel.AddOrUpdateInputParamValueAndCalculatedValue("PrimaryKeyColumn", "ID");
+            actExcel.AddOrUpdateInputParamValueAndCalculatedValue("ColMappingRules", "First='Marco'");
+            actExcel.ExcelActionType = ActExcel.eExcelActionType.WriteData;
+
+            //Act
+            actExcel.Execute();
+
+            //Assert
```

Comment:
Need to validate that the expected values were wrote to the copy/temp excel file by reading it as you wish

Reply:
done